### DISCUSSION TITLE
next js + static file rewrites via vercel.json

### DISCUSSION THREAD

**Discussion by skene** [vercel/vercel#4494] [Help] [answered]
Hello,

I'm hosting a nextjs site on vercel and receiving a 404 when trying to rewrite files in the public folder to a specific url. The vercel.json looks like,

```
{
  "version": 2,
  "rewrites": [
    {
      "source": "/.well-known/(.*)",
      "destination": "/src/public/well-known/$1"
    }
  ]
}
```

My structure looks like
```
src
    components
    pages
    public
         well-known
              file.txt
```

any ideas? Thanks in advanced

**Comment by styfle** [comment, marked as answer]
The problem is that the `/public` directory must be in the root, not `/src/public`.

Once you move the public directory, delete your `vercel.json` because it is not necessary.

Next.js Docs: https://nextjs.org/docs/advanced-features/src-directory
